lib/charms/operator_libs_linux/v0/juju_systemd_notices.py

```diff
@@ -148,6 +149,22 @@ def _systemctl(*args) -> None:
 _disable_service = functools.partial(_systemctl, "disable")
 
 
+@dataclass
+class Service:
+    """Systemd service to observe.
+
+    Args:
+        name: Name of systemd service to observe on dbus.
+        alias: Event name alias for service.
+    """
+
+    name: str
+    alias: Optional[str] = None
+
+    def __post_init__(self) -> None:  # noqa D105
+        self.alias = self.alias or self.name
```

Comment:
Nit: in terms of data modelling, I don't think we should do this. If it has an alias, alias should be set, otherwise None (this also makes the field type correctly optional). Code that uses `Service.alias` should do the fallback, for example `s.alias or s.name`